restdocs-api-spec-jsonschema/src/test/kotlin/com/epages/restdocs/apispec/jsonschema/JsonSchemaFromFieldDescriptorsGeneratorTest.kt

```diff
@@ -289,6 +292,89 @@ class JsonSchemaFromFieldDescriptorsGeneratorTest {
         thenSchemaIsValid()
     }
 
+    @Test
+    fun should_specify_generic_items_type_in_array_when_descriptor_doesnt_contain_itemsType_in_additionalParameters() {
+        givenFieldDescriptorWithTopLevelArrayOfAny()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(CombinedSchema::class.java)
+        val combinedSchema = ((schema as ArraySchema).allItemSchema) as CombinedSchema
+        then(combinedSchema.criterion).isEqualTo(ONE_CRITERION)
+        then(combinedSchema.subschemas).extracting("class").containsExactlyInAnyOrder(
+            ObjectSchema::class.java,
+            BooleanSchema::class.java,
+            StringSchema::class.java,
+            NumberSchema::class.java
+        )
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_specify_accurate_items_type_in_array_when_descriptor_contains_itemsType_in_additionalParameters() {
+        givenFieldDescriptorWithArrayOfSingleType()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(StringSchema::class.java)
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_specify_generic_items_type_in_array_of_array_when_descriptor_doesnt_contain_itemsType_in_additionalParameters() {
+        givenFieldDescriptorWithTopLevelArrayOfArrayOfAny()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(ArraySchema::class.java)
+        val arrayOfArraySchema = ((schema as ArraySchema).allItemSchema) as ArraySchema
+        then(arrayOfArraySchema.allItemSchema).isInstanceOf(CombinedSchema::class.java)
+        val combinedSchema = arrayOfArraySchema.allItemSchema as CombinedSchema
+        then(combinedSchema.criterion).isEqualTo(ONE_CRITERION)
+        then(combinedSchema.subschemas).extracting("class").containsExactlyInAnyOrder(
+            ObjectSchema::class.java,
+            BooleanSchema::class.java,
+            StringSchema::class.java,
+            NumberSchema::class.java
+        )
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_specify_accurate_items_type_in_array_of_array_when_descriptor_contains_itemsType_in_additionalParameters() {
+        givenFieldDescriptorWithTopLevelArrayOfArrayOfSingleType()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(ArraySchema::class.java)
+        val arrayOfArraySchema = (schema as ArraySchema).allItemSchema as ArraySchema
+        then(arrayOfArraySchema.allItemSchema).isInstanceOf(StringSchema::class.java)
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_not_erased_array_content() {
```

Comment:
Did this happen during development? Was this the case before?

I'd formulate it positively:
```
should_keep_mixed_items_schema_for_array_with_mixed_content()
```
Does this match the intention?
Is "mixed" schema the correct terminology? What's the difference to `CombinedSchema` (if it's the same: name it "combined_items_schema" and "combined_content")?

Reply:
Actually, the intention here is to check if the content of an array of objects is well created. At some point in the development some content was change with a `combinedSchema` instead of an `objectSchema`, and all the tests were green.